frontend-wasm/src/module_translator.rs

```diff
@@ -10,16 +10,47 @@ use crate::sections_translator::{
 use crate::wasm_types::FuncIndex;
 use crate::{unsupported_diag, WasmTranslationConfig};
 use miden_diagnostics::DiagnosticsHandler;
-use miden_hir::Module;
+use miden_hir::{FunctionIdent, Module, Program, ProgramBuilder};
 use std::prelude::v1::*;
 use wasmparser::{NameSectionReader, Parser, Payload, Validator, WasmFeatures};
 
-/// Translate a sequence of bytes forming a valid Wasm binary into Miden IR
+/// Translate a sequence of bytes forming a valid Wasm binary into Miden IR module
 pub fn translate_module(
     wasm: &[u8],
-    _config: &WasmTranslationConfig,
+    config: &WasmTranslationConfig,
     diagnostics: &DiagnosticsHandler,
 ) -> WasmResult<Module> {
+    translate_module_inner(wasm, config, diagnostics).map(|res| res.module)
+}
+
+/// Translate a sequence of bytes forming a valid Wasm binary into Miden IR program
+pub fn translate_program(
```

Comment:
FYI: I'm planning to add attributes to HIR entities like `Program`, `Module`, and `Function`, so that we can, for example, decorate a `Function` with an attribute that indicates that it is the entrypoint for whatever `Program` it gets linked into.

I think that will make this function unnecessary, though there may be other reasons to have it for testing and such. In all other situations, this should be handled by `midenc-compile` though I think.

Reply:
I only introduced it to pass the Wasm's start function as an entrypoint, and I did not like how it came out. It turned out kind of useless anyway since we can only get limited `main` function there. I'm going to remove it.